File: Find triplets with zero sum[8-July-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
    #Function to find triplets with zero sum.    
    def findTriplets(self, arr, n):
        arr.sort()
        
        for i in range(n-2):
            j=i+1
            k=n-1
            
            while(j<k):
                sum=arr[i]+arr[j]+arr[k]
                if(sum==0):
                    return 1
                elif sum>0:
                    k-=1
                else:
                    j+=1
                    
        return 0

File: Find triplets with zero sum[8-July-2023]/solution.cpp
//  >> C++ CODE <<

class Solution{
  public:
    //Function to find triplets with zero sum.
    bool findTriplets(int arr[], int n)
    { 
        sort(arr,arr+n);
        
        for(int i=0;i<n;i++)
        {
            int j=i+1;
            int k=n-1;
            
            while(j<k)
            {
                if(arr[i]+arr[j]+arr[k]==0) return true;
                else if(arr[i]+arr[j]+arr[k]>0) k--;
                else j++;
            }
            
        }
        
            return false;
    }
};